pkg/piecestore/rpc/server/api/api.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"golang.org/x/net/context"
+
+	"storj.io/storj/pkg/piecestore"
+	"storj.io/storj/pkg/piecestore/rpc/server/ttl"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+// Server -- GRPC server meta data used in route calls
+type Server struct {
+	PieceStoreDir string
+	DB            *ttl.TTL
+}
+
+// Store -- Store incoming data using piecestore
+func (s *Server) Store(stream pb.PieceStoreRoutes_StoreServer) error {
+	log.Println("Storing data...")
+
+	total := int64(0)
+
+	// Receive initial meta data about what's being stored
+	piece, err := stream.Recv()
+	if err == io.EOF {
+		return err
+	} else if err != nil {
+		return err
```

Comment:
maybe since these branches are both the same, don't need to worry about the explicit EOF check?

Reply:
👍